golubevm/lab6/2/server.c

```diff
@@ -0,0 +1,103 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#define PORT 5000
+#define BUFFER_SIZE 1024
+
+void handle_client(int client_socket) {
+    char buffer[BUFFER_SIZE];
+    ssize_t bytes_read;
+
+    while ((bytes_read = read(client_socket, buffer, BUFFER_SIZE)) > 0) {
+        write(client_socket, buffer, bytes_read);
+    }
+
+    close(client_socket);
+    exit(0);
+}
+
+void config_server(struct sockaddr_in *server_address) {
+    server_address->sin_family = AF_INET;
+    server_address->sin_addr.s_addr = INADDR_ANY;
+    server_address->sin_port = htons(PORT);
+}
+
+void handle_error(const char *msg, int fd) {
+    perror(msg);
+    close(fd);

```

Comment:
> When writing to a file on a remote filesystem, close()-time errors are woefully common, if the local network is prone to glitches or just drops a lot of packets. As an end user, I want my applications to tell me if there was an error when writing to a file. Usually the connection to the remote filesystem is broken altogether, and the fact that writing to a new file failed, is the first indicator to the user.
If you don't check the close() return value, the application will lie to the user. It will indicate (by a lack of an error message if not otherwise), that the file was correctly written, when in fact it wasn't, and the application was told so; the application just ignored the indication. If the user is like me, they'll be very unhappy with the application.